lib/ret.ex

```diff
@@ -24,10 +26,17 @@ defmodule Ret do
         @asset_file_columns
       )
 
-      delete_entities_with_owned_files(
-        from(project in Project, where: project.created_by_account_id == ^account_to_delete.account_id),
-        @project_file_columns
-      )
+      project_query = from(project in Project, where: project.created_by_account_id == ^account_to_delete.account_id)
+      scene_query = from(scene in Scene, where: scene.account_id == ^account_to_delete.account_id)
+
+      project_and_scene_owned_files =
+        collect_owned_files(project_query, @project_file_columns) ++
+          collect_owned_files(scene_query, @scene_file_columns)
+
+      Repo.delete_all(project_query)
+      Repo.delete_all(scene_query)
+
+      delete_owned_files(project_and_scene_owned_files)
```

Comment:
We probably want to make this atomic so that we can completely comply with GDPR, even if a user is adding files as we are deleting their account. The way to do so would be to put all the database operations in a transaction, and only if that transaction is committed to delete the actual files on disk.

I know it’s a pain to rewrite that way. I can help. I’m sorry I didn’t catch it sooner!